tools/preflight/helper.go

```diff
@@ -0,0 +1,448 @@
+}
+
+//  clusterHasVolumeSnapshotClass checks and returns volume snapshot class if present on cluster.
+func clusterHasVolumeSnapshotClass(kubeAccessor *internal.Accessor, snapshotClass string) (*unstructured.Unstructured, error) {
+	vsscList, err := getAllVolumeSnapshotClass(kubeAccessor, schema.GroupVersionKind{
+		Group:   "snapshot.storage.k8s.io",
+		Version: "v1beta1",
+		Kind:    "VolumeSnapshotClass",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, vssc := range vsscList.Items {
+		if vssc.GetName() == snapshotClass {
+			return &vssc, nil
+		}
+	}
+
+	return nil, goerr.New("Volume snapshot class " + snapshotClass + " not found...")
+}
+
+//  createDNSPodSpec returns a corev1.Pod instance.
+func createDNSPodSpec(imagePath string, op *Options) *corev1.Pod {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      dnsUtils + resNameSuffix,
+			Namespace: op.Namespace,
+			Labels: map[string]string{
+				labelTrilioKey:       labelTvkPreflightValue,
+				labelPreflightRunKey: resNameSuffix,
+				labelK8sPartOf:       labelK8sPartOfValue,
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:            dnsContainerName,
+					Image:           imagePath + "/" + dnsUtilsImage,
+					Command:         []string{"sleep", "3600"},
+					ImagePullPolicy: corev1.PullIfNotPresent,
+					Resources: corev1.ResourceRequirements{
+						Requests: map[corev1.ResourceName]resource.Quantity{
+							"memory": *resource.NewQuantity(64*1024*1024, resource.BinarySI),
+							"cpu":    resource.MustParse("250m"),
+						},
+						Limits: map[corev1.ResourceName]resource.Quantity{
+							"memory": *resource.NewQuantity(128*1024*1024, resource.BinarySI),
+							"cpu":    resource.MustParse("500m"),
+						},
+					},
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyAlways,
+		},
+	}
+
+	if op.ImagePullSecret != "" {
+		pod.Spec.ImagePullSecrets = []corev1.LocalObjectReference{
+			{
+				Name: op.ImagePullSecret,
+			},
+		}
+	}
+	if op.ServiceAccountName != "" {
+		pod.Spec.ServiceAccountName = op.ServiceAccountName
+	}
+
+	return pod
+}
+
+func createVolumeSnapshotPVCSpec(storageClass, namespace string) *corev1.PersistentVolumeClaim {
+	pvc := &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sourcePvc + resNameSuffix,
+			Namespace: namespace,
+			Labels: map[string]string{
+				labelTrilioKey:       labelTvkPreflightValue,
+				labelPreflightRunKey: resNameSuffix,
+				labelK8sPartOf:       labelK8sPartOfValue,
+			},
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+			StorageClassName: &storageClass,
+			Resources: corev1.ResourceRequirements{
+				Requests: map[corev1.ResourceName]resource.Quantity{
+					"storage": resource.MustParse("1Gi"),
+				},
+			},
+		},
+	}
+
+	return pvc
+}
+
+func createVolumeSnapshotPodSpec(namespace, pvcName string) *corev1.Pod {
```

Comment:
for all the pods in preflight, try to use same template with common fields and override as required